timApp/util/flask/search.py

```diff
@@ -189,220 +178,164 @@ def to_json(self):
         }
 
 
-class ParResult:
+@dataclass()
+class SearchResult:
     """
-    Paragraph search results.
+    Common superclass for different search results.
     """
 
-    def __init__(
-        self, par_id: str = "", preview: str = "", word_results=None, alt_num_results=0
-    ):
-        """
-        Paragraph result object constructor.
-
-        :param par_id: Paragrapg id.
-        :param preview: A snippet from paragraph markdown.
-        :param word_results: List of word search results in the paragraph.
-        :param alt_num_results: Alternative to listing word results.
-        """
-        if word_results is None:
-            word_results = []
-        self.par_id = par_id
-        self.preview = preview
-        self.word_results = word_results
-        self.alt_num_results = alt_num_results
+    alt_num_results: int = 0
+    word_results: list[WordResult] = field(default_factory=list)
 
     def add_result(self, result: WordResult) -> None:
         """
-        Add new word result.
+        Add new result to the list.
 
-        :param result: New word result from paragraph markdown.
+        :param result: New word result.
         :return: None.
         """
         self.word_results.append(result)
 
     def has_results(self) -> bool:
         """
-        :return: True if the object contains results.
+        :return: Whether the object contains any results.
         """
         return len(self.word_results) > 0 or self.alt_num_results > 0
 
+    def get_match_count(self) -> int:
+        """
+        :return: How many match words the search result has.
+        """
+        if not self.word_results:
+            return self.alt_num_results
+        return len(self.word_results)
+
+    def to_json(self):
+        """
+        :return: A dictionary of attributes and derived attributes, suitable for JSON-conversion.
+        """
+        return {"results": self.word_results, "num_results": self.get_match_count()}
+
+
+@dataclass()
```

Comment:
```suggestion
@dataclass
```

jne (näitä on ilmeisesti aika monta)